exercises/tom_hampson/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,126 @@
+#include "ShapeSorter.h"
+
+#include <algorithm>
+#include <iostream>
+
+ShapeSorter::~ShapeSorter()
+{
+}
+
+/**
+ * @brief Print all shapes with specified shape type (e.g. "circle").
+ * @param shapeType 
+*/
+void ShapeSorter::printShapesOfType(const std::string& shapeType) const
+{
+  std::cout << "Shapes of type \"" << shapeType << "\":" << std::endl;
+
+  for (auto shape : m_shapes)
```

Comment:
```suggestion
  for (const auto& shape : m_shapes)
```
so as not to increase the ref count of each of the shapes (shared pointers)

Reply:
Done